.github/workflows/ci-repo.yml

```diff
@@ -11,71 +11,104 @@ on:
         uses: actions/setup-node@v4
         with:
           node-version-file: "vscode/.nvmrc"
-
+          
       - name: Cache extension node_modules
         uses: actions/cache@v3
         with:
           path: vscode/node_modules
           key: ${{ runner.os }}-node-${{ hashFiles('vscode/package-lock.json') }}
 
       - name: Install dependencies
+        working-directory: ./vscode
         run: |
-          cd vscode
           npm version
           npm ci
 
       - name: Lint sources
-        run: |
-          cd vscode 
-          npm run lint
+        working-directory: ./vscode
+        run: npm run lint
 
       - name: Build
+        working-directory: ./vscode
+        run: npm run package
+
+      # Install Xvfb and Run tests on Linux
+      - name: Install Xvfb and Run tests (Linux)
+        if: matrix.arch == 'linux'
+        working-directory: ./vscode
+        run: 
+          xvfb-run -a npm run test
+
+      # Run tests on macOS
+      - name: Run tests (macOS)
+        if: matrix.arch == 'macos'
+        working-directory: ./vscode
+        run: npm test
+
+      # Run tests on Windows
+      - name: Run tests (Windows)
+        if: matrix.arch == 'windows'
+        working-directory: ./vscode
+        shell: cmd
+        run: npm test
+
+      - name: Generate .vsix package
+        working-directory: ./vscode
         run: |
-          cd vscode
-          npm run package
-
-  vscode-tests:
-    runs-on: ubuntu-latest
-
-    steps:
-      - name: Check-out repository
-        uses: actions/checkout@v4
+          npm install @vscode/vsce
+          npx vsce package
 
-      - name: Use Node.js from .nvmrc
-        uses: actions/setup-node@v4
+      - name: Upload VSIX artifact
+        uses: actions/upload-artifact@v4
         with:
-          node-version-file: "vscode/.nvmrc"
+          name: vscode-extension-${{ matrix.arch }}-${{ github.run_id }}
+          path: "vscode/*.vsix"
 
-      - name: Cache extension node_modules
-        uses: actions/cache@v3
-        with:
-          path: vscode/node_modules
-          key: ${{ runner.os }}-node-${{ hashFiles('vscode/package-lock.json') }}
+      - name: Generate tag name
+        id: vars
+        shell: bash
+        run: |
+          SHORT_SHA=$(echo "${GITHUB_SHA}" | cut -c1-8)
+          TAG_NAME="dev-${SHORT_SHA}"
+          echo "tag_name=$TAG_NAME" >> $GITHUB_OUTPUT
 
-      - run:  |
-            cd vscode
-            npm ci
-      - run: |
-            cd vscode
-            xvfb-run -a npm run test
-      
\ No newline at end of file
+      - name: Create or Update Pre-release
+        if: matrix.arch == 'linux' && github.ref == 'refs/heads/main'
+        uses: ncipollo/release-action@v1
+        with:
+          tag: ${{ steps.vars.outputs.tag_name }}
+          allowUpdates: true
+          commit: ${{ github.sha }}
+          artifacts: "vscode/*.vsix"
+          prerelease: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I think this should be it's own build that:
1. Runs only for `refs/heads/main`
2. `needs: package`
2. Assuming that `tag` hasn't been the source of our problem, then we should make this a semantic version compatible string like `v0.0.1-dev+${{ github.sha }}` (you are welcome to shorten it if it looks terrible).

Reply:
@dzager: do you know where can I get the release version from? do I need to add an env variable or is there any other way to know?

Reply:
@djzager I fetched the version from the package.json but I am not able to test it on my fork ( messed up main branch :( )